File: src/main/java/com/github/tuk-cps/jAADD/iRange.java
package jAADD;
/**
 * The package jAADD implements an Affine Arithmetic Decision Diagram.
 *
 * @author Christoph Grimm, Carna Zivkovic
 */

/**
 * A class that is a range from a lower to an upper bound of the Integers.
 */

public class iRange {
    protected Integer lb, ub;

    // Constructors
    iRange (Integer l, Integer u) {lb = u; ub = u; }

    // Some operations.
    void add(iRange r) {
        lb= r.lb + lb;
        ub= r.ub + ub;
    }

    void join(iRange r) {
        lb = Math.min(r.lb, lb);
        ub = Math.max(r.ub, ub);
    }

    void intersect(iRange r) {
        lb = Math.max(r.lb, lb);
        ub = Math.min(r.ub, ub);
    }

    Boolean isEmptySet() { return lb > ub; }
}

File: src/main/kotlin/com/github/tuk-cps/jAADD/iRange.kt
package jAADD

/**
 * The package jAADD implements an Affine Arithmetic Decision Diagram.
 *
 * @author Christoph Grimm, Carna Zivkovic
 */
/**
 * A class that is a range from a lower to an upper bound of the Integers.
 */
internal class iRange internal constructor(protected var min: Int, protected var max: Int) {

    // Some operations.
    operator fun plus(r: iRange) {
        min = r.min + min
        max = r.max + max
    }

    infix fun join(r: iRange) {
        min = Math.min(r.min, min)
        max = Math.max(r.max, max)
    }

    infix fun intersect(r: iRange) {
        min = Math.max(r.min, min)
        max = Math.min(r.max, max)
    }

    val isEmptySet: Boolean
        get() = min > max

}